File: ai_job_replacement_helper/utils/formatters.py
"""
Formatting utilities
"""

from datetime import datetime
from typing import Optional


def format_date(date: datetime, format: str = "%Y-%m-%d") -> str:
    """Formatear fecha"""
    return date.strftime(format)


def format_currency(amount: float, currency: str = "EUR") -> str:
    """Formatear moneda"""
    if currency == "EUR":
        return f"€{amount:,.2f}"
    elif currency == "USD":
        return f"${amount:,.2f}"
    else:
        return f"{amount:,.2f} {currency}"


def format_duration(seconds: int) -> str:
    """Formatear duración en segundos"""
    if seconds < 60:
        return f"{seconds}s"
    elif seconds < 3600:
        minutes = seconds // 60
        return f"{minutes}m {seconds % 60}s"
    else:
        hours = seconds // 3600
        minutes = (seconds % 3600) // 60
        return f"{hours}h {minutes}m"

File: agents/backend/onyx/server/features/robot_movement_ai/frontend/lib/utils/formatters.ts
/**
 * Formatter utilities - consolidated formatting functions
 */

import { formatNumber, formatCurrency, formatDate, formatDuration, formatFileSize, formatPercentage, formatPosition } from './format';

// Re-export all formatters for convenience
export {
  formatNumber,
  formatCurrency,
  formatDate,
  formatDuration,
  formatFileSize,
  formatPercentage,
  formatPosition,
};

/**
 * Format bytes to human readable
 */
export function formatBytes(bytes: number, decimals: number = 2): string {
  return formatFileSize(bytes);
}

/**
 * Format timestamp to relative time
 */
export function formatRelativeTime(date: Date | string): string {
  const dateObj = typeof date === 'string' ? new Date(date) : date;
  const now = new Date();
  const diff = now.getTime() - dateObj.getTime();
  const seconds = Math.floor(diff / 1000);
  const minutes = Math.floor(seconds / 60);
  const hours = Math.floor(minutes / 60);
  const days = Math.floor(hours / 24);

  if (days > 0) return `hace ${days} día${days > 1 ? 's' : ''}`;
  if (hours > 0) return `hace ${hours} hora${hours > 1 ? 's' : ''}`;
  if (minutes > 0) return `hace ${minutes} minuto${minutes > 1 ? 's' : ''}`;
  return 'hace unos segundos';
}

/**
 * Format number with unit
 */
export function formatWithUnit(value: number, unit: string, decimals: number = 2): string {
  return `${formatNumber(value, decimals)} ${unit}`;
}

/**
 * Format range
 */
export function formatRange(min: number, max: number, unit?: string): string {
  const formatted = `${formatNumber(min)} - ${formatNumber(max)}`;
  return unit ? `${formatted} ${unit}` : formatted;
}



